src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Storage {
+    private File file;
+    private Path path;
+
+    public Storage(String filePath) {
+        if (filePath.isBlank()) {
+            path = Paths.get(System.getProperty("user.dir"), "src", "main", "tasks.txt");
+        } else {
+            path = Paths.get(System.getProperty("user.dir"), filePath);
+        }
+        file = new File(path.toUri());
```

Comment:
Nice implementation of storage using File. Checkout FileInputStream / FileOutputStream which serializes the TaskList object so that you don't have to write so much code to process the data!